Removes DHL eCommerce International, replaces DHL eCommerce with DHL eCommerce Solutions (logos)

FangedParakeet opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Description
<!-- Explain the motivation for making this change -->
This allows DHL merchant carrier accounts to be displayed with the same DHL logo that is used by DHL Express.

### Related issue(s)
Automattic/woocommerce-shipping-issues#125.

<!--
  Is there an issue open this PR addresses? If so, link to it for more information.
  GitHub link example: "Fixes #1234" Syntax: `[close|closes|closed|fix|fixes|fixed|resolve|resolves|resolved] #1234`
  Note: Remove this section if this PR does not have a related issue.
-->

### Steps to reproduce & screenshots/GIFs
<!--
  Please include steps to reproduce, as well as screenshots or GIFs, showing the before & after.
  This helps expedite review and increase confidence for approval & merge.
-->
1. Pull the changes from Automattic/woocommerce-connect-server#1674 and follow the steps to enable DHL Express, DHL eCommerce Solutions, and DHL eCommerce Asia.
2. Visit **WooCommerce > Settings > Shipping > WooCommerceShipping**. You should now see the new DHL carrier accounts appear in the list of available Carrier Accounts, all using the same DHL logo as DHL Express.

![DHL carrier account logos](https://cdn-std.droplr.net/files/acc_1104206/WSXHid)
_New DHL carrier accounts with correct logos_

### Checklist

<!-- All testable code should have tests. -->
- [ ] unit tests

<!-- An entry in the changelog file is always required -->
- [x] `changelog.txt` entry added

frosso reviewed (approved):
Looks straightforward 👍 
The `changelog.txt` entry seems to be "checked", but I'm not seeing changes on the file 👀

FangedParakeet commented:
@frosso, initially when I made these changes to use the same logos for DHL Express, DHL eCommerce, DHL eCommerce International, and DHL eCommerce Asia, I added the following line to the changelog.

> * Add 	- Uses same DHL logo for all registered DHL accounts.

Then pretty much exactly as that change got approved, I found out that EasyPost were actually removing DHL eCommerce and DHL eCommerce International to replace them with DHL eCommerce Solutions without explicitly warning us, prompting this quick update. This is basically additional work required for that last change (but I had to make a new PR, since the previous one had just been merged), so I thought I could avoid crowding the changelog entries with a redunandant row. So when I checked the `changelog.txt` entry, what I told you was true...from a certain point of view. But let me know if you think this little update necessitates an additional `Tweak` line and I'll slip it in there for good measure!

superdav42 commented:
Ran into a bug https://github.com/Automattic/woocommerce-services/issues/2298 but not related to these changes.

FangedParakeet commented on classes/class-wc-rest-connect-shipping-carrier-controller.php:
```diff
@@ -26,6 +26,7 @@ public function post( $request ) {
 			return $error;
 		}
 
+		do_action( 'wc_connect_fetch_service_schemas' );
```
@superdav42, I just added this line to refresh the shipping methods regardless of the value for `WOOCOMMERCE_CONNECT_FREQUENT_FETCH`. Seems a bit too easy: will this do the trick or would you recommend a better way to call `WC_Connect_Service_Schemas_Store::fetch_service_schemas_from_connect_server`?

superdav42 replied:
Using the action is perfect. You just need it add it to `WC_REST_Connect_Shipping_Carrier_Delete_Controller::delete()` as well.

superdav42 reviewed (changes requested):
Refresh carriers when a carrier is disconnected as well.

FangedParakeet commented:
Just added that to `WC_REST_Connect_Shipping_Carrier_Delete_Controller` as well and updated the changelog entry to reflect the recent release. Cheers, @superdav42!

superdav42 reviewed (approved):
Works for me now.
